includes/User.php

```diff
@@ -2683,32 +2683,40 @@ public function getDefaultGlobalPreference( $preference ) {
 	 * @return string
 	 */
 	public function getGlobalAttribute( $attribute, $default = null ) {
-		if ( $this->shouldLogAttribute( $attribute ) ) {
-			$this->logAttribute( $attribute );
+
+		// There are currently 2 attributes we want to get from the attribute
+		// service directly every time. "avatar" and "location". These are attributes
+		// which can be updated by clients other than MW. By talking to the service
+		// we make sure to skip MW's user cache which may have a stale value for
+		// that attribute. Check to see if we should be using the service here before
+		// falling back to the getOptionHelper which uses the user cache.
+		if ( $this->shouldGetAttributeFromService( $attribute ) ) {
+			return $this->userAttributes()->getAttribute( $this->getId(), $attribute, $default );
 		}
 
 		return $this->getOptionHelper($attribute, $default);
 	}
 
 	/**
-	 * Returns true if 1.) User is logged in and 2.) The attribute is one used by clients other
-	 * than MW (eg, the avatar service or discussion app).
 	 * @param $attributeName
 	 * @return bool
 	 */
-	private function shouldLogAttribute( $attributeName ) {
-		return $this->isLoggedIn() && in_array( $attributeName, UserAttributes::$ATTRIBUTES_USED_BY_OUTSIDE_CLIENTS );
-	}
+	private function shouldGetAttributeFromService( $attributeName ) {
+		global $wgEnableReadsFromAttributeService;
 
-	/**
-	 * See SOC-1315 for more details
-	 * @param $attributeName
-	 */
-	private function logAttribute( $attributeName ) {
-		$this->info( 'USER_ATTRIBUTES get_attribute_call', [
-			'attributeName' => $attributeName,
-			'userId' => $this->getId()
-		] );
+		if ( empty( $wgEnableReadsFromAttributeService ) ) {
+			return false;
+		}
+
+		if ( !$this->isLoggedIn() ) {
```

Comment:
Good point @kvas-damian. It looks like this method name is actually misleading. It just calls `$this->getId()` and returns if it's 0. `$this->getId()` says it returns 0 if [the user is anonymous or nonexistent](https://github.com/Wikia/app/blob/dev/includes/User.php#L1860). So it does what we want, but the method name is confusing. I'll update to just use `$this->getId()` to make it more clear.